server/graphql/v2/mutation/TransactionMutations.ts

```diff
@@ -2,16 +2,70 @@ import express from 'express';
 import { GraphQLNonNull, GraphQLString } from 'graphql';
 
 import orderStatus from '../../../constants/order_status';
-import { canReject } from '../../../graphql/common/transactions';
+import { TransactionKind } from '../../../constants/transaction-kind';
+import { FEES_ON_TOP_TRANSACTION_PROPERTIES } from '../../../constants/transactions';
 import { purgeCacheForCollective } from '../../../lib/cache';
 import { notifyAdminsOfCollective } from '../../../lib/notifications';
 import models from '../../../models';
-import { Forbidden, NotFound, Unauthorized } from '../../errors';
+import { canReject } from '../../common/transactions';
+import { Forbidden, NotFound, Unauthorized, ValidationFailed } from '../../errors';
 import { refundTransaction as legacyRefundTransaction } from '../../v1/mutations/orders';
+import { AmountInput, getValueInCentsFromAmountInput } from '../input/AmountInput';
 import { fetchTransactionWithReference, TransactionReferenceInput } from '../input/TransactionReferenceInput';
 import { Transaction } from '../interface/Transaction';
 
 const transactionMutations = {
+  addPlatformTipToTransaction: {
+    type: new GraphQLNonNull(Transaction),
+    description: 'Add platform tips to a transaction',
+    args: {
+      transaction: {
+        type: new GraphQLNonNull(TransactionReferenceInput),
+        description: 'Reference to the transaction in the platform tip',
+      },
+      amount: {
+        type: GraphQLNonNull(AmountInput),
+        description: 'Amount of the platform tip',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<typeof Transaction> {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to add a platform tip');
+      }
+
+      const fundsTransaction = await fetchTransactionWithReference(args.transaction, { throwIfMissing: true });
+
+      if (!req.remoteUser.isAdmin(fundsTransaction.HostCollectiveId)) {
+        throw new Unauthorized('Only host admins can add platform tips');
+      } else if (fundsTransaction.kind !== TransactionKind.ADDED_FUNDS) {
+        throw new ValidationFailed('Platform tips can only be added on added funds at the moment');
+      }
+
+      const isPlatformTipAvailable = await models.Transaction.findOne({
+        where: {
+          TransactionGroup: fundsTransaction.TransactionGroup,
+          kind: 'PLATFORM_TIP',
+        },
+      });
+
+      if (isPlatformTipAvailable) {
+        throw new Error('Platform tip is already set for this transaction group');
+      }
+
+      const transaction = {
+        data: { isFeesOnTop: true },
+        TransactionGroup: fundsTransaction.TransactionGroup,
+        platformFeeInHostCurrency: getValueInCentsFromAmountInput(args.amount),
+        FromCollectiveId: fundsTransaction.HostCollectiveId,
+        kind: TransactionKind.PLATFORM_TIP,
+        ...FEES_ON_TOP_TRANSACTION_PROPERTIES,
```

Comment:
If we don't set `amountInHostCurrency` here, my understanding is that `feeOnTopPercent` will be `NaN`, which can impact other values: https://github.com/opencollective/opencollective-api/blob/606b7c02928f22df8ec7b9269faf6b50778f0023/server/models/Transaction.js#L586

Reply:
@Betree : Great catch; thanks a lot. 😄 I just pushed a change (86d90a4) to set this amount to the platform tip amount. My understanding is that the conversion rate multiplier should be 1 here as the open collective hosts currency is always 'USD'. Let me know if that's not the case.